src/helpers.rs

```diff
@@ -456,6 +456,18 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         }
     }
 
+    /// Dispatches to appropriate implementations for reading an OsString from Memory,
+    /// depending on the interpretation target.
+    /// FIXME: Use `Cow` to avoid copies
+    fn read_os_str_from_target_str(&self, scalar: Scalar<Tag>) -> InterpResult<'tcx, OsString> {
```

Comment:
The `target_str` convention isn't great IMO, how about `target_encoding`? `wide_str`, be it UTF-16 or UCS-2, is an *encoding*, while WTF-8 is another.

Reply:
That's even longer though... `_from_target`?